File: food-recipe-backend/graphql/queries.go
package graphql

import "github.com/hasura/go-graphql-client"

var client = graphql.NewClient("http://localhost:8081/v1/graphql", nil)

type Recipe struct {
	ID           int
	Title        string
	Ingredients  string
	Instructions string
}

var GetRecipesQuery struct {
	Recipes []Recipe `graphql:"recipes"`
}

File: food-recipe-frontend/src/graphql/queries.js
// /src/graphql/queries.js
import { gql } from '@apollo/client/core';

export const GET_RECIPES = gql`
  query GetRecipes {
    recipes {
      id
      title
      ingredients
      instructions
      recipeCategory
    }
  }
`;

export const GET_RECIPE_BY_ID = gql`
  query GetRecipeById($id: ID!) {
    recipe(id: $id) {
      id
      title
      ingredients
      instructions
      recipeCategory
    }
  }
`;
